sklearn/linear_model/_base.py

```diff
@@ -191,6 +211,20 @@ def _rescale_data(X, y, sample_weight):
                                   shape=(n_samples, n_samples))
     X = safe_sparse_dot(sw_matrix, X)
     y = safe_sparse_dot(sw_matrix, y)
+
+    if order is not None:
+        sparse_format = "csc" if order == "F" else "csr"
```

Comment:
I find this somewhat not obvious, when calling `_rescale_data(..., order='C')` on a sparse array. In fact this doesn't need to be part of `_rescale_data` it could be a separate private function enforcing order when needed.

Reply:
I moved this in a private function _set_order in `__coordinate_descent.py`. It's the only place where it is used.